src/geometry/WorkingGeometry.ts

```diff
@@ -130,4 +207,48 @@ export class WorkingGeometry {
             vertexCount += child.vertices.length;
         }
     }
+
+    /**
+     * Special divide for scaling vectors. Treats undeterminate as 0.
+     *
+     * @param {number} a: Numerator.
+     * @param {number} b: Denominator.
+     * @return {number}: Result of division.
+     */
+    private scalingDivide(a: number, b: number): number {
+        if (b === 0 && a === 0) {
+            return 0;
+        } else if (b === 0 || a === 0) {
+            return NaN;
```

Comment:
Another thing here: if we hold the origin, and scale (1, 1, 0) to (2, 0, 0), we'd get a NaN. I guess that would be asking for first a rotation and then a scale or a skew and a scale, since these vectors aren't the same direction, so it makes sense if we don't want to handle this case in this method. But we should probably throw an error or something when the vectors aren't in the same direction, since finding the source of NaNs can be hard to do.

Reply:
That makes sense, I only had in mind initially that we'd try to scale in one direction. The skew case still hasn't been accounted for yet.

Reply:
Added check for things pointing in the same direction. Now throws an error. Also added TODOs in areas of improvement.